### PR TITLE
Allow ECDSA serving certificate

### PR DESCRIPTION
**Pull request by Alkemic**
Why this change:
In our environment we are using ECDSA certificates only, unfortunately by design istio-csr uses only RSA. New flag `--serving-use-ec-algorithm` changes serving certificate to ECDSA, and using `app.server.serving.useECAlgo` in helm adds this flag to k8s deployment and changes private key for `istiod.istio-system.svc` certificate.

### CODE REVIEW

File: cmd/app/options/options.go [hunk 1]
```diff
@@ -200,6 +200,8 @@ func (o *Options) addTLSFlags(fs *pflag.FlagSet) {
 		"serving-certificate-key-size", 2048,
 		"Number of bits to use for the server's serving certificate (RSAKeySize).")
 
+	fs.BoolVar(&o.TLS.ServingUseECAlgorithm, "serving-use-ec-algorithm", false,
+		"Use Elliptical Signature algorithm to use when generating private keys.")
```
Comment (JoshVanL) [inline cmd/app/options/options.go:204]: We should change this to be a string with accepts either `"RSA"` or `"ECDSA"` which defaults to `"RSA"`.

File: deploy/charts/istio-csr/templates/certificate.yaml [hunk 2]
```diff
@@ -34,8 +34,13 @@ spec:
   renewBefore: {{ .Values.app.tls.istiodCertificateRenewBefore }}
   privateKey:
     rotationPolicy: Always
+    {{- if .Values.app.server.serving.useECAlgo }}
+    algorithm: ECDSA
+    size: 256
+    {{- else }}
     algorithm: RSA
     size: {{ .Values.app.tls.istiodPrivateKeySize }}
+    {{- end }}
```
Comment (JoshVanL) [inline deploy/charts/istio-csr/templates/certificate.yaml:43]:
```suggestion
    algorithm: {{ .Values.app.server.serving.algorithm }}
    size: {{ .Values.app.tls.istiodPrivateKeySize }}
```

**Review by JoshVanL — CHANGES_REQUESTED** [review]
Thanks @Alkemic! Can we please make declaring the algorithm to use a string, and default everything to RSA (to keep the default the same behavior as today.

/assign @Alkemic

File: cmd/app/options/options.go (hunk 1 above)
Comment (Alkemic) [reply cmd/app/options/options.go]: Done. Btw: do you think it's worth to add validation to one of two?

Comment (JoshVanL) [reply cmd/app/options/options.go]: Yes please!

**Comment by SgtCoDFish** [comment]
I'd be willing to see this get over the line, but the test failures look legitimate!

**Comment by inteon** [comment]
@SgtCoDFish This PR should be ready to be merged now.

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

Thanks @Alkemic for raising this and @inteon for the final few tweaks!

I haven't tested this thoroughly, but the changes make sense to me in a vacuum so I'm happy to merge :+1:

**Pull request merged**